### PR TITLE
Adding express route port authorization apis

### PR DESCRIPTION
**Pull request by utbarn-ms**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by utbarn** [comment]
@MyronFanQiu @MikhailTryakhov can you guys please review the PR. Thanks.

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRoutePort.json [hunk 1]
```diff
@@ -601,6 +601,252 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
```
Comment (MyronFanQiu) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRoutePort.json:789]: Just curious about it. How do we choose the `final-state-via`? You choose `location` for delete but `azure-async-operation` for `put`. Could you share the reason behind this?

**Review by mmyyrroonn — APPROVED** [review]
LGTM

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRouteCircuit.json [hunk 2]
```diff
@@ -2239,6 +2239,10 @@
         "globalReachEnabled": {
           "type": "boolean",
           "description": "Flag denoting global reach status."
+        },
+        "authorizationKey": {
```
Comment (pilor) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRouteCircuit.json:2243]: is this a secret?  If so it should not be in any responses and be marked as such with http://azure.github.io/autorest/extensions/#x-ms-mutability

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRoutePort.json [hunk 3]
```diff
@@ -884,6 +1130,13 @@
           },
           "description": "Reference the ExpressRoute circuit(s) that are provisioned on this ExpressRoutePort resource."
         },
+        "authorizations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ExpressRoutePortAuthorization"
```
Comment (pilor) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRoutePort.json:1136]: Parents should not contain child resources in their body.  This breaks RBAC and makes policy, etc... harder for users.  Either it is a child resource or it is a property in the port, not both.

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRoutePort.json (hunk 3 above)
Comment (utbarn) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRoutePort.json]: I see our other resources having child resources as a property - is this something we started to mandate recently? - either ways resolved it.

Comment (pilor) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRoutePort.json]: Its a bad pattern that we are trying to get rid of

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRoutePort.json [hunk 4]
```diff
@@ -913,6 +1159,13 @@
         "identity": {
           "$ref": "./network.json#/definitions/ManagedServiceIdentity",
           "description": "The identity of ExpressRoutePort, if configured."
+        },
+        "authorizations": {
```
Comment (pilor) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRoutePort.json:1163]: This shouldn't be here either.  A child resource should not appear in any way inside the parents request/response.  You have separate APIs that should be used to interact with authorizations

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRoutePort.json (hunk 4 above)
Comment (utbarn) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-07-01/expressRoutePort.json]: Got it. Thanks.

**Review by pilor — APPROVED** [review]

**Pull request merged**